server/sonar-server/src/main/java/org/sonar/server/issue/ws/SearchResponseFormat.java

```diff
@@ -322,17 +336,30 @@ private static void formatIssueComments(SearchResponseData data, Issues.Issue.Bu
     return result;
   }
 
-  private Common.Users.Builder formatUsers(SearchResponseData data) {
-    Common.Users.Builder wsUsers = Common.Users.newBuilder();
+  private Users.Builder formatUsers(SearchResponseData data) {
+    Users.Builder wsUsers = Users.newBuilder();
     List<UserDto> users = data.getUsers();
     if (users != null) {
       for (UserDto user : users) {
-        wsUsers.addUsers(commonFormat.formatUser(user));
+        wsUsers.addUsers(formatUser(user));
       }
     }
     return wsUsers;
   }
 
+  public Users.User.Builder formatUser(UserDto user) {
+    Users.User.Builder builder = Users.User.newBuilder()
+      .setLogin(user.getLogin())
+      .setName(nullToEmpty(user.getName()))
+      .setActive(user.isActive());
+    setNullable(user.getEmail(), text -> builder.setAvatar(hash(text)));
+    return builder;
+  }
+
+  private static String hash(String text) {
```

Comment:
why haven't you extracted and factored this code? It's used also used in SearchMembersAction. I'd even make it a pico component to be able to mock it